components/stargate/cmd/client.go

```diff
@@ -0,0 +1,98 @@
+package cmd
+
+import (
+	"github.com/formancehq/stack/components/stargate/internal/client"
+	"github.com/formancehq/stack/components/stargate/internal/client/interceptors"
+	"github.com/formancehq/stack/libs/go-libs/otlp/otlpmetrics"
+	"github.com/formancehq/stack/libs/go-libs/otlp/otlptraces"
+	app "github.com/formancehq/stack/libs/go-libs/service"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"go.uber.org/fx"
+	"google.golang.org/grpc/keepalive"
+)
+
+const (
+	organizationIDFlag = "organization-id"
+	stackIDFlag        = "stack-id"
+
+	stargateServerURLFlag = "stargate-server-url"
+	gatewayURLFlag        = "gateway-url"
+
+	workerPoolMaxWorkersFlag = "worker-pool-max-worker"
+	workerPoolMaxTasksFlag   = "worker-pool-max-tasks"
+
+	ClientChanSizeFlag                = "client-chan-size"
+	HTTPClientTimeoutFlag             = "http-client-timeout"
+	HTTPClientMaxIdleConnsFlag        = "http-client-max-idle-conns"
+	HTTPClientMaxIdleConnsPerHostFlag = "http-client-max-idle-conns-per-host"
+
+	KeepAliveClientParamTimeFlag                = "keepalive-client-param-time"
+	KeepAliveClientParamTimeoutFlag             = "keepalive-client-param-timeout"
+	KeepAliveClientParamPermitWithoutStreamFlag = "keepalive-client-param-permit-without-stream"
+
+	AuthRefreshTokenDurationFlag = "auth-refresh-token-duration"
```

Comment:
The token acquired from the auth server is a JWT with a "exp" claim.
So you could refresh the token a few minutes before the expiration and avoid this configuratoin.